Log/MessageViewController.swift

```diff
@@ -69,7 +73,9 @@ class MessageViewController: UIViewController {
                                 self.friendConversation?.appendMessageToMessageStack(messageObj: messageObj);
                             }
                         }
-                        self.messagesTableView.reloadData();
+                        DispatchQueue.main.async {
+                            self.messagesTableView.reloadData();
+                        }
                         self.messagesTableView.scrollToBottom();
                     }
```

Comment:
scrollToBottom is also need to included in main queue block. All methods which invoke by UI outlets (tableview, label, view, etc) need to guarantee running on main thread. 

Please see the link below:)
https://www.quora.com/Why-must-the-UI-always-be-updated-on-Main-Thread